LSPS3/README.md

```diff
@@ -0,0 +1,668 @@
+  and that error is only allowed at the final hop.
+* MUST fail with `unknown_next_peer` if all parts have not
+  summed up to at least `payment_size` plus `opening_fee` AND
+  `opening_fee_params.valid_until` has passed.
+* MAY fail with `unknown_next_peer` if it receives too many payment
+  parts.
+* When all conditions below are true, MUST continue to the next
+  step:
+  * The current time is strictly less than
+    `opening_fee_params.valid_until`.
+  * All received payment parts sum up to at least `payment_size`
+    plus `opening_fee`.
+  * The client is connected.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+
+The LSP and client MUST negotiate these options and parameters:
+
+* `option_zeroconf` is set.
+* `option_scid_alias` is set.
+* `announce_channel` is `false`.
+
+> **Rationale** `option_zeroconf` is set so that the payment can
+> be immediately forwarded to the client and resolved as soon as
+> the opening negotiation completes; if the client and LSP had to
+> wait for confirmation, the incoming payment would have to be
+> locked for an extended period, which is indistinguishable from
+> a channel jamming attack.
+> `option_zeroconf` cannot be set unless `announce_channel` is
+> `false`.
+> `option_scid_alias` needs to be set so that the channel can be
+> referred to on future invoices before the channel confirms.
+>
+> Future versions of this API may be able to utilize Asynchronous
+> Receive (currently being developed as of this version) to
+> be able to get around the `option_zeroconf` requirement, by
+> treating the receiver / client as offline until the client is
+> online *and* the JIT channel to it is deeply confirmed, if the
+> LSP and client negotiate a non-`option_zeroconf` channel for
+> the JIT channel.
+> A non-`option_zeroconf` JIT channel could also be later
+> published with `announce_channel = true`.
+
+Clients MAY reject the channel open attempt if any other
+parameters or options are not desired by the client (for example,
+it MAY reject non-zero channel reserve requirements imposed on
+the client by the LSP).
+Rejection is via the `error` Lightning BOLT message.
+
+LSPs MUST fail the payment with `unknown_next_hop` in case the
+client rejects the channel open via an `error`.
+LSPs MUST NOT treat client disconnection during the opening
+protocol as a rejection, but MUST retry if the client
+reconnects before the `valid_until` datetime.
+
+Clients MUST NOT expect that `jit_channel_scid` is the same as
+the `alias` SCID.
+Clients MUST NOT expect that `jit_channel_scid` is different from
+the `alias` SCID.
+
+The LSP then forwards parts to the client as soon as both client
+and LSP have signalled `channel_ready` for the new channel.
+
+The LSP generates non-standard forwards, where the amount received
+by the client is smaller than specified in the onion;
+the client MUST accept the non-standard forward.
```

Comment:
Routehint feerates are paid by the payer, not the receiver.  The payer might not appreciate being the one paying for the channel open; it really should be paid by the client, since the client benefits more from it especially if the payer only pays to the client once (the client continues to enjoy use of the channel for succeeding inbound payments, the payer only needs it for this single payment).

Reply:
Also the base part will be overpaid if the payer is able to use a multipath payment.

Reply:
Resolving.